client/components/domains/registrant-extra-info/uk-form.jsx

```diff
@@ -0,0 +1,198 @@
+			STRA: translate( 'UK Sole Trader', {
+				comment: 'Refers to UK legal concept of self-employment/sole proprietorship',
+			} ),
+			PTNR: translate( 'UK Partnership' ),
+			LLP: translate( 'UK Limited Liability Partnership' ),
+			CRC: translate( 'UK Corporation by Royal Charter' ),
+			FCORP: translate( 'Non-UK Corporation' ),
+			IP: translate( 'UK Industrial/Provident Registered Company' ),
+			LTD: translate( 'UK Limited Company' ),
+			PLC: translate( 'UK Public Limited Company' ),
+			SCH: translate( 'UK School' ),
+			GOV: translate( 'UK Government Body' ),
+			RCHAR: translate( 'UK Registered Charity' ),
+			STAT: translate( 'UK Statutory Body' ),
+			OTHER: translate( 'UK Entity that does not fit another category' ),
+			FOTHER: translate( 'Non-UK Entity that does not fit another category' ),
+		};
+		const registrantTypeOptions = map( registrantTypes, ( text, optionValue ) => (
+			<option value={ optionValue } key={ optionValue }>
+				{ text }
+			</option>
+		) );
+
+		this.state = {
+			registrantTypes,
+			registrantTypeOptions,
+		};
+	}
+
+	componentWillMount() {
+		// Add defaults to redux state to make accepting default values work.
+		const neededRequiredDetails = difference(
+			[ 'registrantType' ],
+			keys( this.props.contactDetailsExtra )
+		);
+
+		// Bail early as we already have the details from a previous purchase.
+		if ( isEmpty( neededRequiredDetails ) ) {
+			return;
+		}
+
+		this.props.updateContactDetailsCache( {
+			extra: pick( defaultValues, neededRequiredDetails ),
+		} );
+	}
+
+	handleChangeEvent = event => {
+		this.props.updateContactDetailsCache( {
+			extra: { [ camelCase( event.target.id ) ]: event.target.value },
+		} );
+	};
+
+	isTradingNameRequired( registrantType ) {
+		return includes(
+			[ 'LTD', 'PLC', 'LLP', 'IP', 'RCHAR', 'FCORP', 'OTHER', 'FOTHER', 'STRA' ],
+			registrantType
+		);
+	}
+
+	isRegistrationNumberRequired( registrantType ) {
+		return includes( [ 'LTD', 'PLC', 'LLP', 'IP', 'SCH', 'RCHAR' ], registrantType );
+	}
+
+	renderTradingNameField() {
+		const { contactDetails, translate } = this.props;
+		const { tradingName } = contactDetails.extra;
+		return (
+			<div>
+				<FormFieldset>
+					<FormLabel className="registrant-extra-info__trading-name" htmlFor="trading-name">
+						{ translate( 'Trading Name' ) }
+					</FormLabel>
+					<FormTextInput
+						id="trading-name"
+						value={ tradingName }
+						autoCapitalize="off"
+						autoComplete="off"
+						autoCorrect="off"
+						placeholder={ '' }
+						onChange={ this.handleChangeEvent }
+					/>
+				</FormFieldset>
+			</div>
+		);
+	}
+
+	renderRegistrationNumberField() {
+		const { contactDetails, translate } = this.props;
+		const { registrationNumber } = contactDetails.extra;
+		return (
+			<div>
+				<FormFieldset>
+					<FormLabel
+						className="registrant-extra-info__registration-number"
+						htmlFor="registration-number"
+					>
+						{ translate( 'Registration Number' ) }
```

Comment:
Hi! I've found a possible matching string that has already been translated **66** times:
`translate( 'registration' )` *ES Score: 10*
See 2 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/cctld-uk)

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).